module/VuFind/tests/unit-tests/src/VuFindTest/Auth/OpenIDConnectTest.php

```diff
@@ -0,0 +1,345 @@
+            'issuer' => 'https://openidconnect.provider.url',
+            'jwks_uri' => 'https://openidconnect.provider.url/oauth/jwks',
+        ];
+        $authModule = $this->getOpenIDConnectObject($config);
+        $this->mockResponse(['openidconnectprovider404.json']);
+        $this->assertEquals($expected, $this->callMethod($authModule, 'getProvider'));
+    }
+
+    /**
+     * Get auth module instance
+     *
+     * @param array $config Configuration
+     *
+     * @return OpenIDConnect
+     */
+    protected function getOpenIDConnectObject(array $config = []): OpenIDConnect
+    {
+        $defaultConfig = [
+            'OpenIDConnect' => [
+                'url' => 'openidconnect.provider.url',
+                'client_id' => 'test_cliend_id',
+                'client_secret' => 'test_client_secret',
+            ],
+        ];
+        $session = new SessionContainer();
+        $oidcConfig = empty($config) ? $defaultConfig : $config;
+        $ilsAuthenticator = $this->getMockILSAuthenticator();
+        $this->openid = new OpenIDConnect($session, $oidcConfig, $ilsAuthenticator);
+        return $this->openid;
+    }
+
+    /**
+     * Mock fixture as HTTP client response
+     *
+     * @param string|array|null $fixture Fixture file
+     *
+     * @return void
+     * @throws InvalidArgumentException Fixture file could not be loaded as HTTP response
+     * @throws RuntimeException         Fixture file does not exist
+     */
+    protected function mockResponse(array $fixture): void
+    {
+        $adapter = new TestAdapter();
+        if (!empty($fixture)) {
+            $responseObj = $this->loadResponse($fixture[0]);
+            $adapter->setResponse($responseObj);
+            array_shift($fixture);
+            foreach ($fixture as $f) {
+                $responseObj = $this->loadResponse($f);
+                $adapter->addResponse($responseObj);
+            }
+        }
+
+        $service = new \VuFindHttp\HttpService();
+        $service->setDefaultAdapter($adapter);
+        $this->openid->setHttpService($service);
+    }
+
+    /**
+     * Load response from file
+     *
+     * @param string $filename File name of raw HTTP response
+     *
+     * @return HttpResponse Response object
+     *
+     * @throws InvalidArgumentException Fixture file could not be loaded as HTTP response
+     * @throws RuntimeException         Fixture file does not exist
+     */
+    protected function loadResponse(string $filename): HttpResponse
+    {
+        return HttpResponse::fromString(
+            $this->getFixture("auth/$filename")
+        );
+    }
+
+    /**
+     * Get a mock ILS authenticator
+     *
+     * @return ILSAuthenticator
+     */
+    protected function getMockILSAuthenticator(): ILSAuthenticator
+    {
+        return $this->getMockBuilder(ILSAuthenticator::class)
+            ->disableOriginalConstructor()
+            ->getMock();
```

Comment:
This can be simplified to:

```suggestion
        return $this->createMock(ILSAuthenticator::class);
```

Reply:
Thanks, it looks much better ;) Done